Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -973,13 +974,13 @@ void FastStringifier::append(JSValue value)
 
     if (value.isInt32()) {
         auto number = value.asInt32();
-        auto length = lengthOfIntegerAsString(number);
-        if (UNLIKELY(!hasRemainingCapacity(length))) {
+        auto* cursor = m_buffer + m_length;
```

Comment:
I suggest:
```
char* cursor = reinterpret_cast<char*>(m_buffer) + m_length;
```
Or something like that. Why repeat the `reinterpret_cast`? Maybe we should even just make a local for `m_buffer` that is a `char*` so we have only one `reinterpret_cast`.

Reply:
Looks good, changed.